tests/test_resource.py

```diff
@@ -0,0 +1,106 @@
+
+    def test_dig(self):
+        assert Resource(None, data={
+            "ProcessorSummary": {"State": "Enabled"}
+        }).dig("ProcessorSummary", "State") == "Enabled"
+
+
+class TestExecuteAction:
+    def setup_method(self,):
+        self.connector = mock.Mock(spec=Connector)
+
+    def test_execute_action(self):
+        Resource(self.connector, data={
+            "Actions": {"#ComputerSystem.Reset": {"target": "/reset/"}}
+        }).execute_action("#ComputerSystem.Reset", "payload")
+
+        self.connector.post.assert_called_once_with("/reset/", payload="payload")
+
+    def test_execute_action_multilevel(self):
+        Resource(self.connector, data={
+            "Actions": {"Oem": {"#ComputerSystem.CustomizedReset": {"target": "/reset/"}}}
+        }).execute_action("#ComputerSystem.CustomizedReset", "payload")
+
+        self.connector.post.assert_called_once_with("/reset/", payload="payload")
+
+    def test_execute_invalid_action(self):
+        with pytest.raises(KeyError):
+            Resource(self.connector, data={
+                "Actions": {"#ComputerSystem.Reset": {"target": "/reset/"}}
+            }).execute_action("#ComputerSystem.InvalidAction", "payload")
+
+
+@mock.patch("time.sleep")
+class TestWaitFor:
+    def setup_method(self):
+        self.connector = mock.Mock(spec=Connector)
+        self.connector.get.return_value.status_code = 200
+        self.connector.get.return_value.headers = {}
+
+    def test_wait_for_value(self, mock_sleep):
+        self.connector.get.return_value.json.side_effect = [
+            {"@odata.id": "id", "PowerState": "On"},
+            {"@odata.id": "id", "PowerState": "On"},
+            {"@odata.id": "id", "PowerState": "Off"}
+        ]
+        assert Resource(self.connector, data={
+            "@odata.id": "id",
+            "PowerState": "On"
+        }).wait_for(["PowerState"], "Off") is True
+
+    def test_wait_for_invalid_value(self, mock_sleep):
+        self.connector.get.return_value.json.side_effect = [
+            {"@odata.id": "id", "PowerState": "On"},
+            {"@odata.id": "id", "PowerState": "On"},
+            {"@odata.id": "id", "PowerState": "Fail"}
+        ]
+        with pytest.raises(BlacklistedValueException):
+            Resource(self.connector, data={
+                "@odata.id": "id",
+                "PowerState": "On"
+            }).wait_for(["PowerState"], "Off", blacklisted=("Fail", ))
+
+    def test_wait_for_timeout(self, mock_sleep):
+        self.connector.get.return_value.json.side_effect = cycle([
+            {"@odata.id": "id", "PowerState": "On"},
+            {"@odata.id": "id", "PowerState": "On"},
+            {"@odata.id": "id", "PowerState": "On"}
+        ])
+        with pytest.raises(TimedOutException):
+            Resource(self.connector, data={
+                "@odata.id": "id",
+                "PowerState": "On"
+            }).wait_for(["PowerState"], "Off", blacklisted=("Fail", ), timeout=0.1)
```

Comment:
We could remove "blacklisted" as it's not relevant

Reply:
Removed